### PR TITLE
feat: render a simple markdown dashboard

### LINKED ISSUE #55: Create a "simple" interop dashboard
> As a libp2p/test-plans maintainer, I want the tooling required to define interop test matrixes and retrieve data in a simple form (a flat table of interop results).

the dashboard was discussed in https://github.com/libp2p/test-plans/pull/44#issuecomment-1257077254)
This will support the epic in https://github.com/libp2p/test-plans/issues/53

## Tasks

- [x] Create a poc to demonstrate the feature
    - [x] create tooling to generate pairs of tests,
    - [x] generate a "simple" output, like a markdown table or a csv output,
- [x] Implement native support in Testground
    - [x] https://github.com/testground/testground/pull/1516
- [x] Run this code and CI and figure out a "good" output
    - [x] output to a workflow?
    - [x] output to the README or a status file in the repo?
    - [x] other?
- [x] use this tooling in CI to build & run "every" interop combinations we know about,
    - [x] Implement the "definitive" version for test generation & output generation
    - [x] Implement CI actions
- [ ] In a follow-up issue: create a nicer output (https://github.com/libp2p/test-plans/issues/62) that can be shared on a website (see https://github.com/libp2p/test-plans/issues/53)

### PR DESCRIPTION
**Pull request by laurentsenta**
Closes #55 

This PR adds a dashboard CI rendered in a table like seen in https://github.com/laurentsenta/test-plans/actions/runs/3620746503 and it adds a badge to the readme which links to the latest dashboard run:

[Interop Dashboard](https://github.com/laurentsenta/test-plans/actions/runs/3620746503).

(fail on purpose, for the sake of demonstration)

- add a new action, `generate-dashboard` which loads a CSV file of testground results and outputs a markdown table
- create a `Dashboard` workflow that calls the composer, generates tests, then generates the dashboard
- create an `Update Badge` workflow, which updates the README badge with a link to the latest dashboard (based on @galargh's [PR](https://github.com/libp2p/test-plans/pull/86))

It assumes that a `results.csv` was generated after running the tests generated by the composer.
See: https://github.com/libp2p/test-plans/issues/55#issuecomment-1333474068

## Todo

- [x] Getting the job summary URL from `pl-strflt/job-summary-url-action` doesn't work yet
- [x] Render the badge with failure if not all tests are passing
- [x] Remove the last commit wich drops other actions to help with debugging
- [ ] Remove errors from the demo in `results.csv` so that CI is green again.

### CODE REVIEW

File: .github/workflows/update-badge.yml [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+          run_id: ${{ github.event.workflow_run.id }}
+          run_attempt: ${{ github.event.workflow_run.run_attempt }}
+          job: generate-dashboard
+      - uses: actions/checkout@v3
+      # https://github.com/orgs/community/discussions/26560
+      - run: |
+          git config user.email "41898282+github-actions[bot]@users.noreply.github.com"
+          git config user.name "github-actions[bot]"
+      - run: |
+          GITHUB_JOB_SUMMARY_URL="${{ steps.metadata.outputs.job_summary_url }}"
```
Comment (laurentsenta) [inline .github/workflows/update-badge.yml:38]: @galargh could you give me a hand with using the job summary action?

This variable is always empty at the moment (example run: https://github.com/laurentsenta/test-plans/actions/runs/3620111366/jobs/6101954513)

Comment (galargh) [reply .github/workflows/update-badge.yml:38]: Sure thing! There you go: https://github.com/libp2p/test-plans/pull/88/files/6b6007c2cf2c5496f739707c7c99ae290ed98547#r1039570057

**Comment by MarcoPolo** [comment]
Do we have to commit the files in `dist/`?

**Review by MarcoPolo — CHANGES_REQUESTED** [review]
some small changes. But overall looks good, thanks! I think we're quite close here.

**Comment by laurentsenta** [comment]
@MarcoPolo I noticed you use that code in https://github.com/libp2p/test-plans/blob/ed5893b0942401fe2351fdbd5240dd41f408d926/multidim-interop/src/lib.ts already, is it correct to assume we'll merge everything with https://github.com/libp2p/test-plans/pull/85 and we can close this PR?

**Pull request closed without merging**